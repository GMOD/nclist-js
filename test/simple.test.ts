//@ts-nocheck
import { RemoteFile } from 'generic-filehandle2'
import fetch from 'cross-fetch'

import NCListStore from '../src'
import makeTestServer from './static_server'

let testServer
beforeAll(async () => {
  testServer = await makeTestServer()
})
afterAll(() => testServer.close())

describe('simple data', () => {
  const testCases = [
    [
      'read with generic-filehandle2 RemoteFile with http urls',
      () => ({
        baseUrl: `${testServer.url}/`,
        urlTemplate: 'volvox_genes/{refseq}/trackData.json',
        readFile: url => new RemoteFile(url, { fetch }).readFile(),
      }),
    ],
  ]
  testCases.forEach(([desc, params]) => {
    test(desc, async () => {
      const store = new NCListStore(params())

      const features = []
      for await (const feature of store.getFeatures({
        refName: 'ctgA',
        start: 0,
        end: 50000,
      })) {
        features.push(feature)
      }

      expect(features.length).toBe(1)
      expect(features[0].get('start')).toBe(1049)
      expect(features[0].get('Start')).toBe(1049)
      expect(features[0].get('zonker')).toBe(undefined)
      expect(features).toMatchSnapshot()
    })
  })
})
