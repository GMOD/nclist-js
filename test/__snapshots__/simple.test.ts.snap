// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`simple data > read with generic-filehandle2 RemoteFile with http urls 1`] = `
[
  [
    0,
    1049,
    9000,
    1,
    "EDEN",
    "EDEN",
    "protein kinase",
    "ctgA",
    "example",
    [
      [
        0,
        1299,
        9000,
        1,
        "EDEN.3",
        "EDEN.3",
        "Eden splice form 3",
        "ctgA",
        "example",
        [
          [
            1,
            6999,
            7600,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            1,
            4999,
            5500,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            2,
            7600,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          [
            2,
            2999,
            3300,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          [
            2,
            1299,
            1500,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          [
            1,
            3300,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      [
        0,
        1049,
        9000,
        1,
        "EDEN.1",
        "EDEN.1",
        "Eden splice form 1",
        "ctgA",
        "example",
        [
          [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          [
            1,
            2999,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      [
        0,
        1049,
        9000,
        1,
        "EDEN.2",
        "EDEN.2",
        "Eden splice form 2",
        "ctgA",
        "example",
        [
          [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
        ],
        "mRNA",
      ],
    ],
    "gene",
  ],
]
`;
